### PR TITLE
more expansion fixes for dataset-dag etc

### PR DESCRIPTION
**Pull request by marcmengel**
Fix for #50, along with assorted cleanups for dag and cmd files so all the templates render clean.

### CODE REVIEW

**Comment by marcmengel** [comment]
Okay, with this latest commit, I have been able to to run a dataset dag with SAM with the
following setup:
----------------------------
```
. /grid/fermiapp/products/common/etc/setups.sh

export  EXPERIMENT=samdev
export  SAM_EXPERIMENT=samdev
export  GROUP=fermilab
setup  fife_utils v3_3_2
unsetup jobsub_client
setup ifdhc v2_6_5, ifdhc_config v2_6_5
setup -j sam_web_client v3_2

export IFDH_PROXY_ENABLE=0
export IFDH_TOKEN_ENABLE=1

ups active

fife_launch -c generic.cfg --stage gen
```

**Review by shreyb — APPROVED** [review]
Looks good.  Is it ready to merge?

**Comment by shreyb** [comment]
Closes #50

File: lib/utils.py [hunk 1]
```diff
@@ -133,6 +133,8 @@ def set_extras_n_fix_units(args, schedd_name, proxy, token):
     args["environment"] = newe
     if args["debug"]:
         sys.stderr.write("leaving set_extras... args: %s\n" % repr(args))
+    args["version"] = "beta12"
```
[the remark this answers is not in the document]
Comment (marcmengel) [reply lib/utils.py]: Actually... that should probably be "jobsub_version" not "version"... Although that suffers from the same 
issue of setting it in one place.

**Comment by marcmengel** [comment]
I think its ready.

**Pull request merged**